padl/transforms.py

```diff
@@ -1760,520 +1844,554 @@ def _add_name_to_splits(self, final_splits):
+                    unbatchify_positions=unbatchify_positions)
+        return forward_edges, unbatchify_found, unbatchify_node_dict
+
+    def _generate_postprocess_dict(self,
+                                   current_node=None,
+                                   postprocess_edges=None,
+                                   unbatchify_positions=None,
+                                   ):
+        if postprocess_edges is None:
+            postprocess_edges = defaultdict(dict)
+
+        if current_node is None:
+            unbatchify_nodes = [n_ for n_ in self.edges if isinstance(n_.transform, Unbatchify)]
+            for unbatch_node in unbatchify_nodes:
+                if unbatchify_positions is not None:
+                    unbatch_pos = unbatchify_positions[unbatch_node]
                 else:
-                    final_splits.append(Parallel(s))
+                    unbatch_pos = None
+                postprocess_edges[self.input_node][unbatch_node] = (unbatch_pos, None)
+                postprocess_edges = self._generate_postprocess_dict(current_node=unbatch_node,
+                                                                    postprocess_edges=postprocess_edges,
+                                                                    unbatchify_positions=unbatchify_positions)
+            return postprocess_edges
+
+        for child in self.edges[current_node]:
+            if isinstance(child.transform, Unbatchify):
+                raise SyntaxError("Two Unbatchify in same path is not allowed")
             else:
-                final_splits.append(s)
-
-        res = []
-        for split in final_splits:
-            try:
-                res.append(group(split))
-            except AttributeError:
-                res.append(split)
-        final_splits = res
-
-        self._add_name_to_splits(final_splits)
-        return output_components, final_splits, has_batchify
+                postprocess_edges[current_node][child] = self.edges[current_node][child]
+                postprocess_edges = self._generate_postprocess_dict(current_node=child,
+                                                                    postprocess_edges=postprocess_edges,
+                                                                    unbatchify_positions=unbatchify_positions)
+        return postprocess_edges
 
-    def __call__(self, args):
-        """Call method for Rollout
+    def _pd_splits(self, input_components=0):
+        """Generate splits
 
-        :param args: Argument to call with
-        :return: namedtuple of outputs
+        :param input_components:
+        :return:
         """
-        out = []
-        for i, transform_ in enumerate(self.transforms):
-            try:
-                out.append(transform_._pd_unpack_args_and_call(args))
-            except Exception as err:
-                self._pd_trace_error(i, args)
-                raise err
+        batchify_positions = None
```

Comment:
is `batchify_positions` needed? we are getting another `batchify_positions` as the output of `_generate_preprocess_dict` in the line 2227